Mention Cloud compatibility in three guides

ptgott opened the pull request:
Server Access getting started:
- Remove the instructions to install and set up the Auth
  Service, and add references to main Getting Started guides
  and Cloud signup page.
- Add a prerequisite to have deployed the Auth Service and
  Proxy Service.

EC2 joining guide: make it explicit that the Cloud is not
compatible with this. Also make small style tweaks.

App Access getting started page:
- The guide assumes that you are running the Auth Service,
  Proxy Service, and App Service via the same binary, which
  does not work for Cloud users.

ptgott committed:
Mention Cloud compatibility in three guides

Server Access getting started:
- Remove the instructions to install and set up the Auth
  Service, and add references to main Getting Started guides
  and Cloud signup page.
- Add a prerequisite to have deployed the Auth Service and
  Proxy Service.

EC2 joining guide: make it explicit that the Cloud is not
compatible with this. Also make small style tweaks.

App Access getting started page:
- The guide assumes that you are running the Auth Service,
  Proxy Service, and App Service via the same binary, which
  does not work for Cloud users.

zmb3 commented on docs/pages/application-access/getting-started.mdx:
````diff
@@ -9,85 +9,92 @@ videoBanner: 5Uwhp3IQMHY
 Now start Teleport and point it to the application endpoint:
 
 ```code
 $ sudo teleport start \
````
Should we prefer `teleport app start` instead of `teleport start --roles=app`?

- `--app-uri` becomes `--uri`
- `--app-name` becomes `--name`
- `--roles` can be removed
- `--token` and `--auth-server` remain unchanged

ulysseskan replied:
If we do prefer `teleport app start` here, then these docs become inconsistent with the manual instructions built into Teleport Cloud > Applications > Add Application > Manually, where it says:

`teleport start --roles=app --app-name=[example-app] --app-uri=http://localhost/ --token=[generated-join-token] --auth-server=example.cloud.gravitational.io:443` 

As an aside, I found Teleport Cloud's automatic method of adding app access (via the auto-installer script) super easy.    I bet most people will use this.

ptgott replied:
@zmb3 What do you think? Shall we open a PR against `gravitational/webapps` to change the [help text](https://github.com/gravitational/webapps/blob/2bf39459e741942d305ffe2d179833dc3b5934ac/packages/teleport/src/Apps/AddApp/Manually/Manually.tsx#L109) and we can merge these two changes at the same time?

zmb3 commented on docs/pages/application-access/getting-started.mdx:
```diff
@@ -9,85 +9,92 @@ videoBanner: 5Uwhp3IQMHY
-  ACME challenge to validate certificate requests which only works on port `443`. Make sure your Teleport proxy is accessible on port `443` when using ACME for certificate management.
-</Admonition>
+### Start Teleport
+On the host where you will run the Teleport Application Service, download the latest version of Teleport for your platform from our
```
I'd prefer to be consistent and keep a newline after the header and before the content.

zmb3 commented on docs/pages/application-access/getting-started.mdx:
```diff
@@ -9,85 +9,92 @@ videoBanner: 5Uwhp3IQMHY
 
-Make sure to update `--app-name` and `--app-uri` accordingly if you're using
-your own web application.
+Change `https://teleport.example.com:3080` to the address and port of your Teleport Auth Server. If you are a Teleport Cloud cluster, use your tenant's subdomain, e.g., `mytenant.teleport.sh`. 
```
Does app service support direct dial? If not, this might need to be changed to `the address and port of your Teleport Proxy Server`.

(My crystal ball tells me that allowing `--auth-server` to point to an auth server or a proxy server depending on whether or not you want to use IoT mode will be looked back on as our billion dollar mistake)

zmb3 commented on docs/pages/server-access/getting-started.mdx:
```diff
@@ -28,43 +27,23 @@ This guide introduces some of these common scenarios and how to interact with Te
-- A registered Domain Name with an available subdomain like `tele.example.com`.
-- A Two-Factor Authentication app (such as Authy or Google Authenticator).
+- The Teleport Auth Service and Proxy Service, deployed on your own infrastructure or managed via Teleport Cloud.
+- One host running your favorite Linux environment (such as Ubuntu 20.05, CentOS 8.0-1905, or Debian 10). This will serve as a Teleport Server Access node.
```
```suggestion
- One host running your favorite Linux environment (such as Ubuntu 20.04, CentOS 8.0-1905, or Debian 10). This will serve as a Teleport Server Access node.
```

zmb3 commented on docs/pages/server-access/getting-started.mdx:
```diff
@@ -28,43 +27,23 @@ This guide introduces some of these common scenarios and how to interact with Te
+## Step 1/4. Install Teleport 
 
-   The first instance will host a public-facing Auth Node `tele.example.com` that will serve as the *Bastion Host* (following the recommended *Bastion*-pattern). As such, leave the following ports open:
+1. Create a new instance of your desired Linux distribution (such as Ubuntu 20.05, CentOS 8.0-1905, or Debian 10).
```
```suggestion
1. Create a new instance of your desired Linux distribution (such as Ubuntu 20.04, CentOS 8.0-1905, or Debian 10).
```

zmb3 commented on docs/pages/server-access/getting-started.mdx:
````diff
@@ -115,49 +94,30 @@ This guide introduces some of these common scenarios and how to interact with Te
 
    ```code
    # Let's save the token to a file
    $ sudo tctl tokens add --type=node | grep -oP '(?<=token:\s).*' > token.file
````
Is sudo necessary for cloud users?

ptgott replied:
I've suggested adding a `Details` box explaining tctl usage (including sudo) for Cloud users. If that looks good, we can look into making this an include and adding it to other relevant guides.

zmb3 commented on docs/pages/server-access/getting-started.mdx:
````diff
@@ -115,49 +94,30 @@ This guide introduces some of these common scenarios and how to interact with Te
    ```
 
-   Each Teleport Node can be configured into SSH mode (Teleport Node) and run as an enhanced SSH server. "Node" mode specifies that the Teleport Node will act and join as an SSH server.
+   Each Teleport node can be configured into SSH mode and run as an enhanced SSH server. `--type=node` specifies that the Teleport node will act and join as an SSH server.
````
Isn't Node a "product proper noun" in this case?

ptgott replied:
It is—I must not have realized this when I made these edits.

ptgott committed:
Respond to PR feedback

- Add newlines after headings
- Use teleport app start instead of teleport start --roles=app
- Fix incorrect Ubuntu version number
- Add a Details box explaining tctl usage for Cloud
- Correctly capitalize "Node"
